lib/utils/disk.go

```diff
@@ -37,3 +43,83 @@ func PercentUsed(path string) (float64, error) {
 	ratio := float64(stat.Blocks-stat.Bfree) / float64(stat.Blocks)
 	return Round(ratio * 100), nil
 }
+
+// CanUserWriteTo attempts to check if a user has write access to certain path.
+// It also works around the program being run as root and tries to check
+// the permissions of the user who executed the program as root.
+// This should only be used for string formatting or inconsequential use cases
+// as it's not bullet proof and can report wrong results.
+func CanUserWriteTo(path string) (bool, error) {
+	// prevent infinite loops with a max dir depth
+	var fileInfo os.FileInfo
+	var err error
+
+	for i := 0; i < 20; i++ {
+		fileInfo, err = os.Stat(path)
+		if err == nil {
+			break
+		}
+		if errors.Is(err, fs.ErrNotExist) {
+			path = filepath.Dir(path)
+			continue
+		}
+
+		return false, trace.BadParameter("Failed to find path: %+v", err)
```

Comment:
Nit: Do not capitalize returned error messages.
```suggestion
		return false, trace.BadParameter("failed to find path: %+v", err)
```